eq-author/src/App/page/Logic/Routing/DestinationSelector/RoutingDestinationContentPicker.js

```diff
@@ -144,48 +80,50 @@ export const UnwrappedRoutingDestinationContentPicker = ({
   );
 };
 
+const destinationProps = PropTypes.arrayOf(
+  PropTypes.shape({
+    id: PropTypes.string.isRequired,
+    logicalDestination: PropTypes.string.isRequired,
+  })
+);
+
 UnwrappedRoutingDestinationContentPicker.propTypes = {
+  id: PropTypes.string,
   loading: PropTypes.bool.isRequired,
   data: PropTypes.shape({
     page: PropTypes.shape({
       availableRoutingDestinations: PropTypes.shape({
-        logicalDestinations: PropTypes.arrayOf(
-          PropTypes.shape({
-            id: PropTypes.string.isRequired,
-            logicalDestination: PropTypes.string.isRequired,
-          })
-        ),
-        questionPages: PropTypes.arrayOf(
-          PropTypes.shape({
-            id: PropTypes.string.isRequired,
-            displayName: PropTypes.string.isRequired,
-          })
-        ),
+        logicalDestinations: destinationProps,
+        questionPages: destinationProps,
         sections: PropTypes.arrayOf(
           PropTypes.shape({
-            id: PropTypes.string.isRequired,
-            displayName: PropTypes.string.isRequired,
+            id: PropTypes.string,
+            logicalDestination: PropTypes.string,
           })
         ),
       }),
     }),
   }),
   selected: PropTypes.object, // eslint-disable-line react/forbid-prop-types
   onSubmit: PropTypes.func,
-  id: PropTypes.string,
 };
 
-const RoutingDestinationContentPicker = props => (
-  <Query
-    query={getAvailableRoutingDestinations}
-    variables={{ input: { pageId: props.pageId } }}
-    fetchPolicy="cache-and-network"
-  >
-    {innerProps => (
-      <UnwrappedRoutingDestinationContentPicker {...innerProps} {...props} />
-    )}
-  </Query>
-);
+const RoutingDestinationContentPicker = ({ pageId, ...otherProps }) => {
+  const { loading, data } = useQuery(getAvailableRoutingDestinations, {
+    variables: { input: { pageId: pageId } },
+    fetchPolicy: "cache-and-network",
+  });
+
+  return (
+    <>
+      <UnwrappedRoutingDestinationContentPicker
+        data={data}
+        loading={loading}
+        {...otherProps}
+      />
+    </>
```

Comment:
Is the fragment needed?

Also - do we still need to have two components here? We could just have RoutingDestinationContentPicker and do away with the wrapped/unwrapped distinction now you're using useQuery - or was it to make testing easier?

```suggestion
  return (
      <UnwrappedRoutingDestinationContentPicker
        data={data}
        loading={loading}
        {...otherProps}
      />
```

Reply:
Removed unwrapped too